src/map.cpp

```diff
@@ -175,24 +174,24 @@ void Map::removeTile(uint16_t x, uint16_t y, uint8_t z)
 			}
 		}
 
-		Item* ground = tile->getGround();
+		auto ground = tile->getGround();
 		if (ground) {
 			g_game.internalRemoveItem(ground);
 			tile->setGround(nullptr);
 		}
 	}
 }
 
-bool Map::placeCreature(const Position& centerPos, Creature* creature, bool extendedPos /* = false*/,
+bool Map::placeCreature(const Position& centerPos, std::shared_ptr<Creature> creature, bool extendedPos /* = false*/,
                         bool forceLogin /* = false*/)
 {
 	bool foundTile;
 	bool placeInPZ;
 
-	Tile* tile = getTile(centerPos.x, centerPos.y, centerPos.z);
+	auto tile = getTile(centerPos.x, centerPos.y, centerPos.z);
```

Comment:
Could this be `const auto`
```suggestion
	const auto tile = getTile(centerPos.x, centerPos.y, centerPos.z);
```